### PR TITLE
Added ColabFold documentation

### PR DESCRIPTION
**Pull request by cissoucr**

### CODE REVIEW

File: docs/Software & Tools/ColabFold.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
+
+# Using a Jupyter notebook
+
+Start by accessing JupyterLab using the Unity OnDemand interface:
+
+[https://ood.unity.rc.umass.edu/pun/sys/dashboard/batch_connect/sys/jupyterlab/session_contexts/new](https://ood.unity.rc.umass.edu/pun/sys/dashboard/batch_connect/sys/jupyterlab/session_contexts/new)
+
+Click on the JupyterLab interactive app and fill out the different fields:
```
Comment (ConnorKenyon) [inline docs/Software & Tools/ColabFold.md:8]: Could specify which fields to leave at default too

File: docs/Software & Tools/ColabFold.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+#SBATCH -e slurm-%j.err
+
+module load cudnn/cuda11-8.4.1.50
+module load cuda/11.4.0
+module load miniconda
+source ~/.bashrc
+
+conda activate colabfold
+
+colabfold_batch <path to directory containing MSAs> <path to output directory> --stop-at-score 85 --msa-mode 'MMseqs2 (UniRef+Environmental)'
```
Comment (ConnorKenyon) [inline docs/Software & Tools/ColabFold.md:109]: if these paths need to be specified explicitly, maybe include something about where they might be found.

**Review by ConnorKenyon — APPROVED** [review]

**Review by ConnorKenyon — APPROVED** [review]

File: docs/Software & Tools/ColabFold.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+5. `Modules`: cudnn/cuda11-8.4.1.50 cuda/11.4.0
+
+The fields `CPU thread count` and `Extra arguments for Slurm` can be left blank.
+
+Inside JupyterLab:
+
+* Open the ColabFold.ipynb notebook
+* Choose Python (colabfold) for the kernel
+* Insert your protein sequence next to query_sequence and execute the code in the cell (press SHIFT+ENTER or press the play button in the toolbar above).
+* Run the code in the remaining cells in order to predict the protein structure with the default parameters (see below) and output plots and a visualization of the 3D structure.
```
Comment (lsaloio) [inline docs/Software & Tools/ColabFold.md:23]: Instead of "(see below)" it might be useful to be more specific, such as "(see **Notes** section below)"

**Review by lsaloio — COMMENTED** [review]
This looks great! I've added some comments that might be helpful from a novice perspective, mainly about being more specific and making key information stand out more. Hope this helps!

**Review by lsaloio — COMMENTED** [review]
I think having an actual Introduction heading/section for this info would work well here. See what you think!

**Comment by georgiastuart** [comment]
@cissoucr Can you rename the directory `Software & Tools` to `tools` so the URL is nicer? We can style it in the menu as `Software & Tools` later.

File: docs/tools/ColabFold.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
+# Introduction to ColabFold on Unity
+
+ColabFold [1] is a software developed to accelerate the prediction of protein 3D structures and protein complexes by integrating the fast search algorithm MMSeqs2 with AlphaFold2 [2] or RoseTTAFold.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:3]:
```suggestion
ColabFold [1] is a software developed to accelerate the prediction of protein 3D structures and protein complexes by integrating the fast search algorithm `MMSeqs2` with AlphaFold2 [2] or RoseTTAFold. 
```

File: docs/tools/ColabFold.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
+# Introduction to ColabFold on Unity
+
+ColabFold [1] is a software developed to accelerate the prediction of protein 3D structures and protein complexes by integrating the fast search algorithm MMSeqs2 with AlphaFold2 [2] or RoseTTAFold. 
+ColabFold is available on Unity by using a Jupyter notebook or a batch script. Both methods make use of one graphics processing unit (GPU) and the AlphaFold2 AI tool. The output includes the predicted protein structure in a PDB format text file along with files to evaluate the results.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:4]:
```suggestion
ColabFold is available on Unity through a Jupyter notebook or a batch script. Both methods make use of one graphics processing unit (GPU) and the AlphaFold2 AI tool. The output includes the predicted protein structure in a PDB format text file along with files to evaluate the results. 
```

File: docs/tools/ColabFold.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
+# Introduction to ColabFold on Unity
+
+ColabFold [1] is a software developed to accelerate the prediction of protein 3D structures and protein complexes by integrating the fast search algorithm MMSeqs2 with AlphaFold2 [2] or RoseTTAFold. 
+ColabFold is available on Unity by using a Jupyter notebook or a batch script. Both methods make use of one graphics processing unit (GPU) and the AlphaFold2 AI tool. The output includes the predicted protein structure in a PDB format text file along with files to evaluate the results. 
+The notebook is designed to run ColabFold with one protein sequence while the batch script can be used to make predictions for multiple protein sequences at once.
+ColabFold on Unity is currently available in a beta version. For questions, please send an email to hpc@umass.edu.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:6]:
```suggestion
ColabFold on Unity is currently available in a beta version. For questions, please send an email to [hpc@umass.edu](mailto:hpc@umass.edu). 
```

File: docs/tools/ColabFold.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+ColabFold [1] is a software developed to accelerate the prediction of protein 3D structures and protein complexes by integrating the fast search algorithm MMSeqs2 with AlphaFold2 [2] or RoseTTAFold. 
+ColabFold is available on Unity by using a Jupyter notebook or a batch script. Both methods make use of one graphics processing unit (GPU) and the AlphaFold2 AI tool. The output includes the predicted protein structure in a PDB format text file along with files to evaluate the results. 
+The notebook is designed to run ColabFold with one protein sequence while the batch script can be used to make predictions for multiple protein sequences at once.
+ColabFold on Unity is currently available in a beta version. For questions, please send an email to hpc@umass.edu. 
+
+# Using a Jupyter notebook to access ColabFold
+
+Start by accessing JupyterLab using the Unity OnDemand interface:
+
+[https://ood.unity.rc.umass.edu/pun/sys/dashboard/batch_connect/sys/jupyterlab/session_contexts/new](https://ood.unity.rc.umass.edu/pun/sys/dashboard/batch_connect/sys/jupyterlab/session_contexts/new)
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:12]:
```suggestion
Start by accessing JupyterLab using the [Unity OnDemand interface](https://ood.unity.rc.umass.edu/pun/sys/dashboard/batch_connect/sys/jupyterlab/session_contexts/new).
```

File: docs/tools/ColabFold.md [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+
+Click on the JupyterLab interactive app and fill out the following fields:
+
+1. The `Partition` field indicates the type of compute nodes to run your interactive session on. One of the gpu partitions should be selected to run ColabFold Jupyter notebook (gpu, gpu-long, uri-gpu or gpu-preempt).
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:16]:
```suggestion
1. The `Partition` field indicates the type of compute nodes to run your interactive session on. One of the gpu partitions should be selected to run ColabFold Jupyter notebook (gpu, gpu-long, uri-gpu or gpu-preempt). For more information on partitions, see [the partition list](https://docs.unity.rc.umass.edu/technical/partitionlist.html).
```

File: docs/tools/ColabFold.md [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+2. The `Maximum job duration` field defines how long the interactive session with JupyterLab should run for. This field can be left with the default value of one hour (1:00:00) for short protein sequences but should be increased to make predictions on larger protein sequences.
+3. The `Memory (in GB)` field defines the amount of memory in gigabytes allocated to your interactive session. To give you an idea of how much memory you may need, 8GB is enough for a protein of 59 amino acids but 50 GB is required for a large protein of 2894 amino acids.
+4. The `GPU count` field is the number of GPUs allocated to your interactive session. It should be set to 1 since ColabFold only runs on a single GPU.
+5. The `Modules` field corresponds to a list of modules to load. The two following modules should be added (separated only by a space) to this field in order to use the GPU: cudnn/cuda11-8.4.1.50 cuda/11.4.0
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:20]:
```suggestion
5. The `Modules` field corresponds to a list of modules to load. The two following modules should be added (separated only by a space) to this field in order to use the GPU: `cudnn/cuda11-8.4.1.50 cuda/11.4.0`
```

File: docs/tools/ColabFold.md [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+
+The fields `CPU thread count` and `Extra arguments for Slurm` can be left blank.
+
+Inside JupyterLab:
+1. Open the ColabFold.ipynb notebook.
+2. Choose Python (colabfold) for the kernel.
+3. Insert your protein sequence next to query_sequence and execute the code in the cell (press SHIFT+ENTER or press the play button in the toolbar above).
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:27]:
```suggestion
3. Insert your protein sequence next to `query_sequence` and execute the code in the cell (press `SHIFT+ENTER` or press the play button in the toolbar above).
```

File: docs/tools/ColabFold.md [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+2. The `Maximum job duration` field defines how long the interactive session with JupyterLab should run for. This field can be left with the default value of one hour (1:00:00) for short protein sequences but should be increased to make predictions on larger protein sequences.
+3. The `Memory (in GB)` field defines the amount of memory in gigabytes allocated to your interactive session. To give you an idea of how much memory you may need, 8GB is enough for a protein of 59 amino acids but 50 GB is required for a large protein of 2894 amino acids.
+4. The `GPU count` field is the number of GPUs allocated to your interactive session. It should be set to 1 since ColabFold only runs on a single GPU.
+5. The `Modules` field corresponds to a list of modules to load. The two following modules should be added (separated only by a space) to this field in order to use the GPU: cudnn/cuda11-8.4.1.50 cuda/11.4.0
+
+The fields `CPU thread count` and `Extra arguments for Slurm` can be left blank.
+
+Inside JupyterLab:
+1. Open the ColabFold.ipynb notebook.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:25]: Is this located somewhere people can copy it to their work directory on Unity?

Comment (georgiastuart) [reply docs/tools/ColabFold.md:25]: Could we put it in the dataset directory and give instructions for copying it via the colabfold interface?

File: docs/tools/ColabFold.md [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+2. Choose Python (colabfold) for the kernel.
+3. Insert your protein sequence next to query_sequence and execute the code in the cell (press SHIFT+ENTER or press the play button in the toolbar above).
+4. Run the code in the remaining cells in order to predict the protein structure with the default parameters (see Notes section below) and output plots and a visualization of the 3D structure.
+5. The output directory containing the results will be located in the folder where you put the ColabFold.ipynb notebook.
+
+#### Notes:
+* ColabFold's notebook is setup to run with the following parameters that can be adjusted by the user:
+    * No templates
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:33]: What does this mean?

File: docs/tools/ColabFold.md [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+    * Number of models: 5
+    * Stop predictions at score 100
+    * Msa mode: mmseqs2_uniref_env
+    * Model type: alphafold2_ptm
+* The Jupyter notebook made available here is a modified version of the AlphaFold2_mmseqs2 notebook [3].
+
+# Using a batch script to run ColabFold
+
+The example of batch scripts provided below should be considered for users dealing with a large number of protein sequences.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:42]:
```suggestion
The example batch scripts provided below are useful for users with a large number of protein sequences.
```

File: docs/tools/ColabFold.md [hunk 14]
```diff
@@ -0,0 +1,275 @@
...
+
+## ColabFold on Unity has 3 main components:
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:44]:
```suggestion
## ColabFold components

ColabFold on Unity has three main components:
```

File: docs/tools/ColabFold.md [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+
+1. A script **colabfold_search** that searches the ColabFold databases of proteins using MMseqs2 to build diverse multiple sequence alignments in A3M format.
+2. A script **colabfold_batch** designed to predict protein structures.
+3. A set of protein databases:
+    * **UniRef30**: database containing 30% sequence identity clustered proteins based on UniRef100 non-redundant protein sequence database. 
+    * **Environmental database** (also called ColabFoldDB): combination of** **the** **Big Fantastic Database (BFD) and the MGnify database with redundancy reduced in addition to metagenomic protein catalogs containing eukaryotic proteins, phage catalogs and an updated version of MetaClust.
+    * **Templates database** (PDB70): database containing 70% sequence identity clustered proteins from the** **Protein Data Bank (PDB) database.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:51]: Can we add a note about where these dbs are located on Unity?

File: docs/tools/ColabFold.md [hunk 16]
```diff
@@ -0,0 +1,275 @@
...
+
+The example of batch scripts provided below should be considered for users dealing with a large number of protein sequences.
+
+## ColabFold on Unity has 3 main components:
+
+1. A script **colabfold_search** that searches the ColabFold databases of proteins using MMseqs2 to build diverse multiple sequence alignments in A3M format.
+2. A script **colabfold_batch** designed to predict protein structures.
+3. A set of protein databases:
+    * **UniRef30**: database containing 30% sequence identity clustered proteins based on UniRef100 non-redundant protein sequence database. 
+    * **Environmental database** (also called ColabFoldDB): combination of** **the** **Big Fantastic Database (BFD) and the MGnify database with redundancy reduced in addition to metagenomic protein catalogs containing eukaryotic proteins, phage catalogs and an updated version of MetaClust.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:50]:
```suggestion
    * **Environmental database** (also called ColabFoldDB): combination of the Big Fantastic Database (BFD) and the MGnify database with redundancy reduced in addition to metagenomic protein catalogs containing eukaryotic proteins, phage catalogs and an updated version of MetaClust.
```

File: docs/tools/ColabFold.md [hunk 17]
```diff
@@ -0,0 +1,275 @@
...
+    * **Templates database** (PDB70): database containing 70% sequence identity clustered proteins from the** **Protein Data Bank (PDB) database.
+
+## Getting started
+### Search against the ColabFold databases
+
+Finding homologous proteins using MMSeqs2 can be done by running a batch script using the SBATCH command: `sbatch <path to batch script file>`
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:56]:
```suggestion
Finding homologous proteins using MMSeqs2 can be done by running a batch script using the SBATCH command: `sbatch <path to batch script file>`. For more information on using `sbatch`, see [our documentation](https://docs.unity.rc.umass.edu/slurm/sbatch.html).
```

File: docs/tools/ColabFold.md [hunk 18]
```diff
@@ -0,0 +1,275 @@
...
+
+The code below is an example of a batch script to run MMSeqs2. The top of the script contains the instruction to use bash to execute the commands (`#!/bin/bash`) and the SBATCH parameters (`#SBATCH <parameter>`) followed by the different modules required to run MMSeqs2 through ColabFold.
+
+
+```
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:61]:
```suggestion
```bash
```

File: docs/tools/ColabFold.md [hunk 19]
```diff
@@ -0,0 +1,275 @@
...
+
+module load uri
+module load MMseqs2/14-7e284-gompi-2021b
+module load miniconda
+source ~/.bashrc
+conda activate colabfold
+```
+
+
+The command colabfold_search shown below is added after activating the conda environment `colabfold-v1.5.0`. In this case, protein sequences contained in a fasta file are aligned against the UniRef30 (`--db1 uniref30_2202/uniref30_2202_db`) and environmental (`--db3 colabfold_envdb_202108/colabfold_envdb_202108_db`) databases. UniRef30 is the default database used to search proteins. In order to use the environmental database, the parameter `--use-env `has to be set to 1 in addition to providing the path (`--db3 colabfold_envdb_202108/colabfold_envdb_202108_db`).
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:79]:
```suggestion
The command `colabfold_search` shown below is added after activating the conda environment `colabfold-v1.5.0`. In this case, protein sequences contained in a fasta file are aligned against the UniRef30 (`--db1 uniref30_2202/uniref30_2202_db`) and environmental (`--db3 colabfold_envdb_202108/colabfold_envdb_202108_db`) databases. UniRef30 is the default database used to search proteins. In order to use the environmental database, the parameter `--use-env `has to be set to 1 in addition to providing the path (`--db3 colabfold_envdb_202108/colabfold_envdb_202108_db`).
```

File: docs/tools/ColabFold.md [hunk 20]
```diff
@@ -0,0 +1,275 @@
...
+colabfold_search <path to fasta file> /datasets/bio/colabfold <path to output directory> --db1 uniref30_2202/uniref30_2202_db --db3 colabfold_envdb_202108/colabfold_envdb_202108_db --use-env 1 --use-templates 0
+```
+
+
+To use the PDB70 templates database, the parameter `--use-templates` should be set to 1 and the path to the database should be provided with `--db2 pdb`.
+
+#### Notes:
+* `<path to fasta file>` is the full path to a fasta file containing protein sequence(s) of interest.
+* `<path to output directory>` is the full path to an existing directory used to store the multiple sequence alignments (MSAs).
+* > Note that it is recommended to request at least 200G using `#SBATCH --mem=200G` in order to load the protein databases.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:92]:
```suggestion
* Note that it is recommended to request at least 200G using `#SBATCH --mem=200G` in order to load the protein databases.
```

File: docs/tools/ColabFold.md [hunk 21]
```diff
@@ -0,0 +1,275 @@
...
+* Running colabfold_search with 1,762 proteins, the UniRef30 and environmental databases and the highest mmseqs sensitivity (s = 8) on a gpu A100 node with 64 threads takes approximately 3h.
+
+### Train models and make predictions with ColabFold using a batch script
+
+A batch script can be used to train models and make predictions with ColabFold. It should be noted that predictions on proteins longer than 1000bp should be run on a GPU node with at least 40GB VRAM and that the whole process can be expedited on a large set of input protein sequences by submitting the batch script as an array job.
+
+The code below provides an example on how to make predictions using `colabfold_batch` in a batch script:
+
+The parameter --stop-at-score is used to stop generating models until the predicted confidence metric (pLDDT or predicted local distance difference test) is reached.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:101]:
```suggestion
The parameter `--stop-at-score` is used to stop generating models until the predicted confidence metric (pLDDT or predicted local distance difference test) is reached.
```

File: docs/tools/ColabFold.md [hunk 22]
```diff
@@ -0,0 +1,275 @@
...
+
+#### Notes:
+* `<path to fasta file>` is the full path to a fasta file containing protein sequence(s) of interest.
+* `<path to output directory>` is the full path to an existing directory used to store the multiple sequence alignments (MSAs).
+* > Note that it is recommended to request at least 200G using `#SBATCH --mem=200G` in order to load the protein databases.
+* Running colabfold_search with 1,762 proteins, the UniRef30 and environmental databases and the highest mmseqs sensitivity (s = 8) on a gpu A100 node with 64 threads takes approximately 3h.
+
+### Train models and make predictions with ColabFold using a batch script
+
+A batch script can be used to train models and make predictions with ColabFold. It should be noted that predictions on proteins longer than 1000bp should be run on a GPU node with at least 40GB VRAM and that the whole process can be expedited on a large set of input protein sequences by submitting the batch script as an array job.
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:97]:
```suggestion
A batch script can be used to train models and make predictions with ColabFold. It should be noted that predictions on proteins longer than 1000bp should be run on a GPU node with at least 40GB VRAM and that the whole process can be expedited on a large set of input protein sequences by submitting the batch script as an [array job](https://slurm.schedmd.com/job_array.html).
```

File: docs/tools/ColabFold.md [hunk 23]
```diff
@@ -0,0 +1,275 @@
...
+
+The code below provides an example on how to make predictions using `colabfold_batch` in a batch script:
+
+The parameter --stop-at-score is used to stop generating models until the predicted confidence metric (pLDDT or predicted local distance difference test) is reached.
+
+
+```
```
Comment (georgiastuart) [inline docs/tools/ColabFold.md:104]:
```suggestion
```bash
```

**Review by georgiastuart — COMMENTED** [review]
Looks really good! All my edits are minor. I think after you resolve them, this is good to go.

**Review by georgiastuart — APPROVED** [review]
Looks great!

**Pull request merged**